tests/testthat/test-helpers-vector.R

```diff
@@ -51,10 +38,21 @@ test_that("any_of() is lax", {
 })
 
 test_that("all_of() and any_of() check their inputs", {
-  expect_error(select_loc(letters2, all_of(NA)), "missing")
-  expect_error(select_loc(letters2, any_of(NA)), "missing")
-  expect_error(select_loc(letters2, all_of(na_chr)), "missing")
-  expect_error(select_loc(letters2, any_of(na_chr)), "missing")
-  expect_error(select_loc(letters2, all_of(TRUE)), class = "vctrs_error_subscript_type")
-  expect_error(select_loc(letters2, any_of(TRUE)), class = "vctrs_error_subscript_type")
+  expect_snapshot(error = TRUE, {
```

Comment:
Could you use `expect_error()` within `expect_snapshot()` to avoid weakening these tests until r-lib/testthat#1667 is solved please?